disktools.py

```diff
@@ -592,9 +592,7 @@ def createPartition(self, id, sizeBytes=None, number=None, order=None, startByte
         partitions = [None] + [part for num, part in sorted(self.partitions.items(), key=lambda item: item[1]['start'])]
 
         if startBytes is None:
-            if len(partitions) == 0:
```

Comment:
Without this check (and with the default [None] value for partitions) won't line 603 fail if it's ever reached? Since partitions[-1] will be None and then the lookup for ["start"] and ["size"] will fail?

Reply:
That would happen with or without this test.

Reply:
Ah you're right, I forgot to consider the None when checking the length! :doh: